rules-java/api/src/main/java/org/jboss/windup/rules/apps/mavenize/MavenizationService.java

```diff
@@ -0,0 +1,422 @@
+
+        // 3) Write the pom.xml's.
+        new MavenStructureRenderer(mavCtx).createMavenProjectDirectoryTree();
+    }
+
+
+    /**
+     * Mavenizes the particular project module, i.e. single pom.xml.
+     * Then continues recursively to submodules.
+     *
+     * @return null if the project can't be processed for some reason, e.g. is from an unparsable jar.
+     */
+    private Pom mavenizeModule(MavenizationContext mavCtx, ProjectModel projectModel, Pom containingModule)
+    {
+        LOG.info("Mavenizing submodule ProjectModel " + projectModel.toPrettyString());
+        LOG.info("    Root file: " + projectModel.getRootFileModel().toPrettyString());
+        LOG.info("    Containing module: " + containingModule);
+
+        if (projectModel.getRootFileModel().getParseError() != null)
+            return null;
+
+        // Known library -> add as a dependency and skip.
+
+        // SHA1 identified archive?
+        if (projectModel.getRootFileModel() instanceof IdentifiedArchiveModel)
+        {
+            final IdentifiedArchiveModel idArch = (IdentifiedArchiveModel)projectModel.getRootFileModel();
+            if (idArch == null)
+                LOG.warning("Project's IdentifiedArchiveModel getRootFileModel() returned null.");
+            else if (idArch.getCoordinate() == null)
+                LOG.warning("Project's IdentifiedArchiveModel getRootFileModel().getCoordinate() returned null.");
+            else if(containingModule == null)
+                LOG.warning("containingModule is null."); // IllegalStateEx?
+            else
+                containingModule.dependencies.add(new SimpleDependency(Dependency.Role.LIBRARY, MavenCoord.from(idArch.getCoordinate())));
+
+            LOG.info("Known library, skipping recursive mavenization: " + idArch.getCoordinate());
+            return null;
+        }
+
+        // ArchiveModel's organizations contain a known org, like Apache.
+        if (projectModel.getRootFileModel() instanceof ArchiveModel)
+        {
+            Set<String> skipOrganizations = new HashSet(Arrays.asList("Apache Sun Iona IBM Codehaus Spring Sonatype JBoss Oracle".toLowerCase().split("")));
+
+            final ArchiveModel arch = (ArchiveModel)projectModel.getRootFileModel();
+            for (OrganizationModel org : arch.getOrganizationModels())
+            {
+                if (skipOrganizations.contains(org.getName().toLowerCase()))
+                {
+                    LOG.info("Library from 3rd party vendor ("+org.getName()+"), skipping recursive mavenization: " + arch.getFilePath());
+                    return null;
+                }
+            }
+        }
+
+        // A MavenProject.
+        // TODO: This covers both app modules and libraries with pom.xml inside. Need to diferentiate - skip only libraries.
+        // TODO: This should disappear after WINDUP-981
+        if (false && projectModel instanceof MavenProjectModel)
```

Comment:
Eclipse sees this as dead code. I read TODO, but don't know why it still here ;)